code/datums/status_effects/debuffs/screen_blur.dm

```diff
@@ -18,7 +18,8 @@
 		return FALSE
 
 	// Refresh the blur when a client jumps into the mob, in case we get put on a clientless mob with no hud
-	RegisterSignal(owner, COMSIG_MOB_LOGIN, PROC_REF(update_blur))
+	owner.RegisterSignals(COMSIG_MOB_LOGIN, SIGNAL_ADDTRAIT(TRAIT_SIGHT_BYPASS), SIGNAL_REMOVETRAIT(TRAIT_SIGHT_BYPASS), PROC_REF(update_blur))
```

Comment:
Why are we doing `owner.RegisterSignals` and why are we trying to register signals on `COMSIG_MOB_LOGIN`, a string

```suggestion
	RegisterSignals(owner, list(COMSIG_MOB_LOGIN, SIGNAL_ADDTRAIT(TRAIT_SIGHT_BYPASS), SIGNAL_REMOVETRAIT(TRAIT_SIGHT_BYPASS)), PROC_REF(update_blur))
```

Reply:
my brain mixed up how add_trait vs add_traits works with signals